### ISSUE TITLE
Crash in `unwrap()` on `MaximizedState`

### ISSUE THREAD

**Issue by ids1024** [pop-os/cosmic-comp#1008]
I've been using emacs and Firefox stacked in one workspace, and have been having an issue lately where setting that up causes the compositor to crash.

```

**Comment by ids1024** [comment]
Okay, I think I (mostly) see what's going on now. After more testing.

* The window is opened and mapped into the tiling layer
* `Shell::maximize_request()` adds it to the floating layer with `map_maxmized`, but doesn't unmap it from the tiling layer
   - Given the comment elsewhere in the code `should still be mapped in tiling`, this seems to be intentional?
* `FloatingLayout::toggle_stacking()` calls `convert_to_stack` to make the `CosmicMapped` a stack
   - But it is still mapped into the stacking layer as a window, not a stack
* `Workspace::set_tiling` moves the stack into the floating layer
   - So now the tiling layer has two `CosmicMapped`, that are maximized, one of which is a stack and one of which is the window (that is also part of the stack)
* `Workspace::set_tiling` now tries to move both elements back to the tiling layer
   - This works for the stack, but fails for the window mapped element with the above unwrap panic
   
I'm not sure exactly why the `maximized_state` of the window `CosmicMapped` is `None` here, but presumably having both a window and a stack containing that window mapped into a layout at the same time is not correct.

**Comment by Drakulix** [comment]
>     * `Shell::maximize_request()` adds it to the floating layer with `map_maxmized`, but doesn't unmap it from the tiling layer
>       
>       * Given the comment elsewhere in the code `should still be mapped in tiling`, this seems to be intentional?

Yes it is. This is the only case where we ever map a window twice.

>     * `FloatingLayout::toggle_stacking()` calls `convert_to_stack` to make the `CosmicMapped` a stack

Doesn't actually happen, but rather we enumerate the tiling layer first in `toggle_stacking_focused`, so it ends up calling `TilingLayout::toggle_stacking()`, but that essentially does the same thing.

The problem with `convert_to_stack` is, that it converts the `CosmicMapped` in place. Almost everything in there is reference-counted, except the `CosmicWindowInternal`, which this function changes. So we end up with two de-synced `CosmicMapped`-elements that share the same `MaximizedState`.


Thus this function should never be called when a double-map is in place. I modified `toggle_stacking_focused` for this reason to first unmaximize and then (potentially) re-maximize the window: https://github.com/pop-os/cosmic-comp/pull/1060

Seems to work.
